File: backend/app/models/repayment.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class RepaymentBase(BaseModel):
    loan_id: str
    amount: float
    payment_date: datetime
    notes: Optional[str] = None

class RepaymentCreate(RepaymentBase):
    pass

class Repayment(RepaymentBase):
    id: str
    user_id: str
    created_at: datetime

    class Config:
        orm_mode = True

class Summary(BaseModel):
    total_loans: int
    total_borrowed: float
    total_repaid: float
    outstanding_amount: float
    next_payment_due: Optional[datetime] = None

File: frontend/src/types/repayment.ts
export interface Repayment {
  id: string;
  loan_id: string;
  user_id: string;
  amount: number;
  payment_date: string;
  notes?: string;
  created_at: string;
}

export interface RepaymentFormData {
  loan_id: string;
  amount: number;
  payment_date: string;
  notes?: string;
}

export interface Summary {
  total_loans: number;
  total_borrowed: number;
  total_repaid: number;
  outstanding_amount: number;
  next_payment_due?: string;
}